### ISSUE TITLE
-Inf for low energy evaluation within gammapy.spectrum.models Absorption

### ISSUE THREAD

**Issue by TarekHC** [gammapy/gammapy#1323]
Hi all, 

I just spotted a simple bug within the Absorption class.

```python
from gammapy.spectrum.models import Absorption
from astropy import units as u
dominguez = Absorption.read_builtin('dominguez').table_model(0.2)
# This works properly, giving 1.0
print(dominguez.evaluate(1 * u.GeV, 1.))
# This doesn't, giving -Inf
print(dominguez.evaluate(1 * u.MeV, 1.))
```

I could fix it easily by putting a "if energy < 1 * u.GeV then return 1.0" within the evaluate method, but there is probably a more pythonic / expert way for doing this...

If you are ok with a quick fix, I could take care of it.

**Comment by cdeil** [comment]
@TarekHC - Yes, please send a PR with a fix.

Ideally, please start by adding a regression test, i.e. test case where you get `nan` now, and a good value after the fix.

I did not look at  http://docs.gammapy.org/dev/_modules/gammapy/spectrum/models.html#Absorption yet to locate where the issue occurs or how best to fix it. Ideally not an if statement, but instead changing the code where we call into numpy or scipy to avoid the `nan`.

@TarekHC - Maybe you could have a look, and if it's not clear how best to fix it, I'll have a look and try to propose something.

PR: @jjlk  wrote that code, but I don't think he has time to review or work on it.

**Comment by adonath** [comment]
This issue has been fixed in the process of #1710. The output is now `1.0` in both cases.
